rotkehlchen/db/upgrade_manager.py

```diff
@@ -75,6 +77,50 @@
 ]
 
 
+class UpgradeProgressHandler():
+    def __init__(
+            self,
+            messages_aggregator: 'MessagesAggregator',
+            target_db_version: int,
+    ) -> None:
+        self._messages_aggregator = messages_aggregator
+        self._target_db_version = target_db_version
+        self._start_db_version: Optional[int] = None
+        self._current_db_version: Optional[int] = None
+        self._current_upgrade_total_steps = 0
+        self._current_upgrade_current_step = 0
+
+    def new_upgrade(self, upgrade_from_version: int) -> None:
+        if self._start_db_version is None:
+            self._start_db_version = upgrade_from_version
+
+        self._current_db_version = upgrade_from_version
+        self._current_upgrade_total_steps = 0
```

Comment:
Note that this logic is zero index based and these numbers will be directly consumed by the frontend so sync with them to see how they woudl like it.

As you have it right now it's gonna be 0/6 until 6/6.. So 0/6 is start. 1/6 is after first upgrade function and 6/6 is at the end. Perhaps that's fine.

Also ... not sure if needed or overkill but we could have a step description for each step. Essentially the name of the function (see v35 -> v36 upgrade). But not sure. PErhaps overkill and not needed. Let's discuss with frontend.

Reply:
As long as it doesn't stop at 5/6 we are fine with that.
As for the description, we could add it but if it comes from the backend it means we can't have i18n for it. 

Also, for the users I don't think it matters that match but it might be useful for debugging.

Reply:
If you think it does not matter for the users then we don't need it. I thought it may be cool to show "work details" to the user, but yes the i8n would be annoying.

No it won't stop at 5/6. I checked that. So let's keep it simple and not have a description at least now.